com.unity.ml-agents/Runtime/Sensors/Reflection/EnumReflectionSensor.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using UnityEngine;
+
+namespace Unity.MLAgents.Sensors.Reflection
+{
+    internal class EnumReflectionSensor: ReflectionSensorBase
+    {
+        Array m_Values;
+
+        internal EnumReflectionSensor(ReflectionSensorInfo reflectionSensorInfo)
+            : base(reflectionSensorInfo, GetEnumObservationSize(reflectionSensorInfo.GetMemberType()))
+        {
+            m_Values = Enum.GetValues(reflectionSensorInfo.GetMemberType());
+        }
+
+        internal override void WriteReflectedField(ObservationWriter writer)
+        {
+            // Write the enum value as a one-hot encoding.
+            // If it's not one of the enum values, put a 1 in the last "bit"
+            var enumValue = GetReflectedValue();
+
+            var knownValue = false;
+            int i = 0;
+            foreach(var val in m_Values)
+            {
+                if (val.Equals(enumValue))
+                {
+                    writer[i] = 1.0f;
+                    knownValue = true;
+                }
+                else
+                {
+                    writer[i] = 0.0f;
+                }
+                i++;
+            }
+
+            writer[i] = knownValue ? 0.0f : 1.0f;
```

Comment:
This means that the dimension of an enum will be n + 1 where n is the number of possible values.
This might be confusing to users and I think returning an array of zeros when the value was none of the possible ones is fair.